apps/condo/domains/ai/tasks/executeAIFlow.js

```diff
@@ -0,0 +1,131 @@
+
+const { FlowiseAdapter } = require('@condo/domains/ai/adapters')
+const {
+    TASK_STATUSES,
+    FLOW_ADAPTERS: FLOW_ADAPTER_NAMES,
+    AI_FLOWS_CONFIG,
+    CUSTOM_FLOW_TYPES_LIST,
+} = require('@condo/domains/ai/constants')
+const { ExecutionAIFlowTask } = require('@condo/domains/ai/utils/serverSchema')
+const { TASK_WORKER_FINGERPRINT } = require('@condo/domains/common/constants/tasks')
+
+const { FLOW_META_SCHEMAS, CUSTOM_FLOW_TYPE } = require('../constants')
+
+
+const BASE_ATTRIBUTES = { dv: 1, sender: { dv: 1, fingerprint: TASK_WORKER_FINGERPRINT } }
+
+const FLOW_ADAPTERS = {
+    [FLOW_ADAPTER_NAMES.FLOWISE]: new FlowiseAdapter(),
+}
+
+const ajv = new Ajv()
+
+const taskLogger = getLogger('executeAIFlow')
+
+const executeAIFlow = async (taskId) => {
+    if (!taskId) {
+        taskLogger.error({
+            msg: 'Unknown taskId!',
+        })
+        throw new Error('Unknown taskId!')
+    }
+
+    const { keystone: context } = getSchemaCtx('ExecutionAIFlowTask')
+
+    const task = await ExecutionAIFlowTask.getOne(context, { id: taskId }, 'id flowType context locale status')
+
+    try {
+        if (!task || task.deletedAt) {
+            throw new Error(`No task with id "${taskId}"`)
+        }
+
+        if (task.status !== TASK_STATUSES.PROCESSING) {
+            throw new Error(`Task with id "${taskId}" has status "${task.status}" already!`)
+        }
+
+        const isCustomFlow = CUSTOM_FLOW_TYPES_LIST.includes(task.flowType)
+        const adapterConfig = AI_FLOWS_CONFIG?.[isCustomFlow ? 'custom' : 'default']?.[task.flowType]
+        if (!adapterConfig) throw new Error(`Cannot AI flow adapter config for flow "${task.flowType}"!`)
+
+        const adapterName = adapterConfig.adapter
+        if (!adapterName) throw new Error(`Unknown AI flow adapter for flow "${task.flowType}"!`)
+
+        const adapter = FLOW_ADAPTERS[adapterName]
+        if (!adapter) throw new Error(`Unexpected AI flow adapter "${adapterName}"!`)
+        if (!adapter.isConfigured) throw new Error(`Adapter "${adapterName}" not configured!`)
+
+        const predictionUrl = adapterConfig.predictionUrl
+        if (!predictionUrl) throw new Error(`Unknown prediction url for flow "${task.flowType}"!`)
+
+        const fullContext = {
+            ...task.context,
+            locale: task.locale,
+        }
+
+        const prediction = await adapter.predict(predictionUrl, fullContext)
+
+        const schema = FLOW_META_SCHEMAS[isCustomFlow ? CUSTOM_FLOW_TYPE : task.flowType]?.output ?? { type: 'object' }
+        const validatePrediction = ajv.compile(schema)
+
+        if (!validatePrediction(prediction.result)) {
+            const validationErrors = validatePrediction.errors.map(error => ({
+                message: error.message,
+                path: error.instancePath,
+            }))
+
+            taskLogger.error({
+                msg: 'Failed to execute AI flow',
+                data: { id: taskId, flowType: task?.flowType },
```

Comment:
maybe we should add taskId near msg so that is indexed correctly

Reply:
updated